src/main/java/com/easypost/service/EasyPostClient.java

```diff
@@ -189,4 +208,41 @@ public String getApiVersion() {
     public String getApiBase() {
         return apiBase;
     }
+
+    /**
+     * Execute an HTTP request for a given method and URL with this EasyPostClient object.
+     *
+     * @param method   the HTTP method to use
+     * @param endpoint the API endpoint to use
+     * @param params   the parameters to use
+     * @param clazz    the class to use for deserializing the response
+     * @param <T>      the type of object to return
+     * @return the deserialized response
+     */
+    public <T> T request(final Requestor.RequestMethod method, final String endpoint, final Map<String, Object> params,
+                         final Class<T> clazz)
+            throws GatewayTimeoutError, RateLimitError, InvalidRequestError, NotFoundError, TimeoutError, EncodingError,
+            UnauthorizedError, MethodNotAllowedError, InternalServerError, UnknownApiError, ServiceUnavailableError,
+            ForbiddenError, JsonError, HttpError, RedirectError, PaymentError {
```

Comment:
We should either just throw `EasyPostException` here since it's a parent class of all the Errors here. If we decide to keep it, you need to add the docstring for all the exceptions. This also apply to below overload function too